lib/jsdom/utils.js

```diff
@@ -161,11 +161,28 @@ exports.memoizeQuery = function memoizeQuery(fn) {
 * A slightly-more-compliant version of `url.resolve`, taking care of a few Node bugs.
 */
 exports.resolveHref = function resolveHref(baseUrl, href) {
-  // If href is "about:blank", Node tries to be too clever.
-  if (href === "about:blank") {
+  var about = "about:blank";
+
+  // if we're redirecting to another site on about:blank, just let it through
+  if (href.substring(0, about.length) === about) {
```

Comment:
Since 3.x is the last version we support node on I wanted to make this compatible so we can issue a last bugfix release for it.